tools/spectral/ipa/rulesets/functions/eachCustomMethodMustUseCamelCase.js

```diff
@@ -0,0 +1,28 @@
+import { getCustomMethod, isCustomMethod } from './utils/resourceEvaluation.js';
+import { hasException } from './utils/exceptions.js';
+import { casing } from '@stoplight/spectral-functions';
+
+const RULE_NAME = 'xgen-IPA-109-custom-method-must-use-camel-case';
+const ERROR_MESSAGE = 'The custom method must use camelCase format.';
+const ERROR_RESULT = [{ message: ERROR_MESSAGE }];
+
+export default (input, opts, { path }) => {
+  // Extract the path key (e.g., '/a/{exampleId}:method') from the JSONPath.
+  let pathKey = path[1];
+
+  if (!isCustomMethod(pathKey)) return;
+
+  if (hasException(input, RULE_NAME)) {
+    return;
+  }
+
+  let methodName = getCustomMethod(pathKey);
+  if (!methodName) {
+    return;
+  }
```

Comment:
Can `methodName` be undefined here if we have a path, for example `/resource:`? If yes this should be an error, as there is nothing after the `:`

Reply:
Yes, it should be an error but I could not decide if it should be validated under this rule. But `camelCase` requires a string length of more than 1, so we can include it I think 👍